src/librustc/infer/error_reporting/mod.rs

```diff
@@ -1312,6 +1321,22 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         }
     }
 
+    /// When encountering a coercion to a supertrait object is attempted,
+    /// note that this is permitted with the "trait upcasting" feature enabled.
+    fn note_enable_trait_upcasting_where_appropriate(
+        &self,
+        exp_found: &ty::error::ExpectedFound<DefId>,
+        diag: &mut DiagnosticBuilder<'tcx>,
+    ) {
+        use rustc::ty::{Binder, TraitRef};
+
+        let trait_ref = Binder::bind(TraitRef::identity(self.tcx, exp_found.found));
+        let supertraits = crate::traits::supertraits(self.tcx, trait_ref);
+        if supertraits.into_iter().any(|trait_ref| trait_ref.def_id() == exp_found.expected) {
+            diag.note("enable the `trait_upcasting` feature to permit upcasting of trait objects");
```

Comment:
```suggestion
            diag.note("add `#![feature(trait_upcasting)]` to the crate attributes to enable");
```
A bit more conventional message. :)

Reply:
Sure.